eliot-service/eliot/cache.py

```diff
@@ -38,8 +38,7 @@ class DiskCache:
 
     """
 
-    # NOTE(willkg): this puts all files in the same directory--no directory trees
-    BAD_CHARS = re.compile(r"[^A-Za-z0-9._-]")
+    BAD_CHARS = re.compile(r"[^A-Za-z0-9._/-]")
```

Comment:
I had to add `/` here. I waffled on whether to use `os.sep` so the system didn't have one more reason it couldn't work on Windows, but that gets tricky for a bunch of reasons. I ended up going with this because we run Eliot in a Docker container anyhow and that's the only runtime environment I'm supporting currently.